libraries/ESP8266WiFi/src/ESP8266WiFiMulti.cpp

```diff
@@ -1,267 +1,507 @@
+            if (WiFi.RSSI(known[j]) > WiFi.RSSI(known[i])) {
+                int8_t tmp;
+
+                // Swap indices
+                tmp = known[i];
+                known[i] = known[j];
+                known[j] = tmp;
+            }
+        }
+    }
+
+    // Print sorted indices
+    DEBUG_WIFI_MULTI("[WIFIM] Sorted indices: ");
+    for (int8_t i = 0; i < numNetworks; i++) {
+        DEBUG_WIFI_MULTI("%d ", known[i]);
+    }
+    DEBUG_WIFI_MULTI("\n");
+
+    // Connect to known WiFi AP's sorted by RSSI
+    for (int8_t i = 0; i < numNetworks; i++) {
+        // Get network information
+        WiFi.getNetworkInfo(known[i], ssid, encType, rssi, bssid, channel, hidden);
+
+        for (auto entry : _APlist) {
+            // Check SSID
+            if (ssid == entry.ssid) {
+                DEBUG_WIFI_MULTI("[WIFIM] Connecting %s\n", ssid);
+
+                // Connect to WiFi
+                WiFi.begin(ssid, entry.passphrase, channel, bssid);
+
+                // Wait for status change
+                if (waitWiFiConnect(connectTimeoutMs) == WL_CONNECTED) {
+                    return WL_CONNECTED;
+                }
+            }
+        }
+    }
+
+    DEBUG_WIFI_MULTI("[WIFIM] Could not connect\n", ssid);
+
+    // Coult not connect to any WiFi network
+    return WL_CONNECT_FAILED;
+}
+
+/**
+ * @brief Print WiFi scan results
+ * @details
+ *      Macro DEBUG_ESP_WIFI and DEBUG_ESP_PORT must be configured
+ */
+void ESP8266WiFiMulti::printWiFiScan()
+{
+#ifdef DEBUG_ESP_WIFI
+    String ssid;
+    int32_t rssi;
+    uint8_t encryptionType;
+    uint8_t* bssid;
+    int32_t channel;
+    bool hidden;
+    int8_t scanResult;
+
+    scanResult = WiFi.scanComplete();
+
+    DEBUG_WIFI_MULTI("[WIFIM] %d networks found:\n", scanResult);
+
+    // Print unsorted scan results
+    for (int8_t i = 0; i < scanResult; i++) {
+        bool known = false;
+
+        WiFi.getNetworkInfo(i, ssid, encryptionType, rssi, bssid, channel, hidden);
+
+        for(auto entry : _APlist) {
+            if(ssid == entry.ssid) {
+                // SSID match
+                known = true;
+            }
+        }
+
+        if (known) {
+            DEBUG_WIFI_MULTI(" --->");
+        } else {
+            DEBUG_WIFI_MULTI("     ");
+        }
+
+        DEBUG_WIFI_MULTI(" %d: [CH %02d] [%02X:%02X:%02X:%02X:%02X:%02X] %ddBm %c %s\n",
+                         i,
+                         channel,
+                         bssid[0], bssid[1], bssid[2],
+                         bssid[3], bssid[4], bssid[5],
+                         rssi,
+                         (encryptionType == ENC_TYPE_NONE) ? ' ' : '*',
+                         ssid.c_str());
+        delay(0);
+    }
+#endif
+}
+
+/**
+ * @brief Print WiFi status
+ * @details
+ *      Macro DEBUG_ESP_WIFI and DEBUG_ESP_PORT must be configured
+ * @param status
+ *      WiFi status
+ */
+void ESP8266WiFiMulti::printWiFiStatus(wl_status_t status)
```

Comment:
This method doesn't depend on anything from ESP8266WiFiMulti. Please remove it from the class and convert it to a standalone function that is private to the cpp (static or inside anonymous namespace)